eth/downloader/downloader.go

```diff
@@ -231,7 +236,10 @@ func New(checkpoint uint64, stateDb ethdb.Database, mux *event.TypeMux, chain Bl
 		quitCh:         make(chan struct{}),
 		SnapSyncer:     snap.NewSyncer(stateDb, chain.TrieDB().Scheme()),
 		stateSyncStart: make(chan *stateSync),
+		syncStartBlock: chain.CurrentFastBlock().NumberU64(),
+		syncStartTime:  time.Now(),
```

Comment:
It can be inaccurate. The start time is assigned once the downloader is constructed, but real sync can happen later.

For example, Geth is ready and waiting for sync trigger, but consensus layer doesn't send us signal for it.
Or the node has been running for very long time(a few days) and a reorg happens in consensus layer, which leads to a tiny sync

In these cases, the estimate time can all be inaccurate.

Reply:
True, but it's a bit inaccurate in general since the chain weight is not linear. But I guess we can init start time only on the first actual "import", then at least it avoid this specific problem.